frontend/pages/campaigns/[campaignId]/index.tsx

```diff
@@ -39,56 +44,119 @@ export default function CampaignDetail() {
   const numTotalCoupons = campaign.promisedAmount / campaign.couponDenomination;
   const numCouponsRedeemed = campaign.donations.primaryDonation.amount / campaign.couponDenomination;
 
+  const getPromisedAmountStatsCardChild = (): ReactNode => (
+    <Typography variant="h4">
+      will be donated to{' '}
+      <strong>
+        {campaign.charities.length} {campaign.charities.length === 1 ? 'charity' : 'charities'}
+      </strong>{' '}
+      when the campaign ends. This was sponsored by {campaign.primaryDonor.name}.
+    </Typography>
+  );
+
+  const getDonationAmountStatsCardChild = (): ReactNode => {
+    if (campaign.donations.secondaryDonation.amount === 0) {
+      return (
+        <Typography variant="h4">
+          No donations have been made yet. <strong>Be the first to donate!</strong> Click the contribute button above to
+          make a donation.
+        </Typography>
+      );
+    }
+
+    return (
+      <Typography variant="h4">
+        extra donations were generated by the public. That makes up{' '}
+        <strong>{Math.round((campaign.donations.secondaryDonation.fraction ?? 0) * 100)}%</strong> of the total donated
+        amount!
+      </Typography>
+    );
+  };
+
+  const getNumCouponsRedeemedStatsCardChild = (): ReactNode => {
+    if (numCouponsRedeemed === 0) {
+      return (
+        <Typography variant="h4">
+          No coupons have been redeemed yet. Let&apos;s wait for the first coupon to be redeemed!
+        </Typography>
+      );
+    }
+
+    if (numCouponsRedeemed === numTotalCoupons) {
+      return (
+        <Typography variant="h4">
+          All coupons have been redeemed! You can still donate to the campaign if you wish to by clicking the contribute
```

Comment:
Side note, Idk if the contribution button will be displayed for non-active campaigns. Depends on the status of #308